### ISSUE TITLE
[Bug]: Open and Save Dialogs stopped working on MAS version

### ISSUE THREAD

**Issue by synapse** [electron/electron#32465]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

12.0.2

### What operating system are you using?

macOS

### Operating System Version

macOS Catalina 10.15.7

### What arch are you using?

x64

### Last Known Working Electron version

12

### Expected Behavior

When running `dialog.showSaveDialog` or `dialog.showOpenDialog` should open the dialog windows for choosing the path

### Actual Behavior

Nothing is happening. `dialog.showMessageBox` seems to be working fine.

### Additional Information

Previously I was using this 3rd party library to access the `remote` object on the render side and it stopped working when the app was downloaded from the Mac App Store. I've switched to `ipcRenderer.invoke` and `ipcMain.handle` and running the dialogs directly from the `main` process but still nothing.

Demo video using the MAS app: https://imgur.com/TstKHJB
URL for the stand-alone version here: https://stacks.rocks

**Comment by VerteDinde** [comment]
Hey @synapse, thanks for reporting this. I looked into a few other Electron apps downloaded through the App Store, but wasn't able to repro your issue.

You mentioned that you were previously using a 3rd party library to access the remote module - would it be possible for you to share the code that you're using for the dialog APIs? Since I can't reproduce this in other Electron apps, it's a bit hard to debug. Sharing that code may help us better see what the issue is.

The only other thing that stands out to me: you're using a deprecated version of Electron, and I know that MAS private API requirements were recently changed. Does the issue persist if you upgrade to the newest version of Electron 13?

I'm adding the blocked/need-info label for now, but I'll remove that once you share code and/or try the newer Electron version.

**Comment by synapse** [comment]
Hey, @VerteDinde I was previously using `@electron/remote` to access the `dialog` API. In fact, I thought it might have some issues and passed to IPC instead, but still the same.

**Comment by synapse** [comment]
So after days of trial and error and more than 5 releases on the Mac App Store I finally found the problem. The good news is it's not Electron.
I'm using these tools to pack and ship my app to the MAS.
- [electron-builder](https://www.electron.build/)
- [electron-osx-sign](https://github.com/electron/electron-osx-sign)
- `productbuild`

While running this command here: `codesign -d --entitlements :- '/app/path/here.app'` I found that for some reason (maybe a new update or something), `electron-osx-sign` was replacing my entitlements in the app (while `electron-builder` was installing correctly) thus removing these two lines which are essential to make the open and save dialogs work:

```
<true/>
    <key>com.apple.security.files.user-selected.read-write</key>
<true/>
    <key>com.apple.security.files.user-selected.read-only</key>
<true/>
```

Anyways to make it work I had to manually specify the (`parent`) entitlement when running `electron-osx-sign` like this:

```
electron-osx-sign "/app/path/here.app" --entitlements="/app/path/entitlement.plist"
```

Sorry for the false alarm @VerteDinde
